docs/adr/0007-validation-guideliness.md

```diff
@@ -35,28 +35,28 @@ attributes in the middle of function execution.
 
 ## Pros, Cons, and Considerations of the Options
 
+Here is how all of our options compare against our requirements:
+
+| Number      | Requirement | ValidationMixin | pydantic | marshmallow |
+| ----------- | ----------- | ----------- | ----------- | ----------- |
+| 1 | Validate function args everywhere | Limited | ✓ | Limited |
+| 2 | Custom deeper validation | ✓ | ✓ | ✓ |
+| 3 | Performance overhead | Minimal. | [Fastest](https://pydantic-docs.helpmanual.io/benchmarks/) | [Slower](https://pydantic-docs.helpmanual.io/benchmarks/) |
+| 4 | Number of new depedencies | 0 | 2 | 1  |
```

Comment:
Thinking of what we would need to vendor into pip, it might be useful if we quantify the new dependencies with the number of files the contain and lines of code in the aggregated dependencies.

Reply:
Added additional commit b486d24 with that information counting lines and files from both direct new dependencies like `pydantic`, `marshmallow` and `typeguard` and transitive dependencies like `typing_extenstions`.